### DISCUSSION TITLE
[Feature] Drag to zoom

### DISCUSSION THREAD

**Discussion by bo0tzz** [immich-app/immich#1712] [Feature Request]
### The feature

Double tap on an image, then drag up and down to zoom in and out.

### Platform

- [ ] Server
- [ ] Web
- [X] Mobile

**Comment by Grunskin** [comment]
This is something I would love to see.

I just got Immich up and running the other day so I'm still trying it out to see if I want to switch.

Double tap and drag-to-zoom is one of the best one-handed features there is in other photo apps (like Google Photos)
I'll try to clarify, you double tap on an image and the last tap you hold on the screen and drag up/down to zoom in/out.
Pinch-to-zoom is mostly a two-handed feature, or really awkward with one hand, and double tap and drag is way better for one-handed operations.

**Comment by varofla** [comment]
I think it would be very helpful if the drag to zoom feature also included an option to reverse the drag direction, especially on iOS. (drag up/down to zoom in/out <-> drag up/down to zoom out/in)

The difference between the drag to zoom direction in iOS’s default apps (like Maps) and in many third-party apps is really confusing for me.

**Comment by theboomr** [comment]
I'm setting up Immich for the first time and did not realize this functionality wasn't present. I know it sounds silly but, while this feature in a photo gallery app isn't quite a deal breaker, it is basically as close to a deal breaker as something can get for me without actually being one. I value one handed ergonomics extremely highly for phone apps so this is very important to me, and I hope it gets implemented soon!

Now if only I had enough free time to find out how to code this and contribute it myself...

**Comment by lividhen** [comment]
Not a super common feature but one I use in almost every app that has it. I'd love to see it here.

https://github.com/user-attachments/assets/cd2e81b0-339f-4e14-9226-10b8fa0667b9
